requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c

```diff
@@ -0,0 +1,51 @@
+/* Body file for function Controller
+ * Generated by TASTE (kazoo/templates/skeletons/c-body/function.tmplt)
+ * You can edit this file, it will not be overwritten
+
+    !! IMPORTANT                                                        !!
+    !! When you modify your design (interface view), you must update    !!
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "controller.h"
+
+#include <CbiPartitionApi.h>
+
+// Implemented in manager.c
+extern void print_rtems(const char *str);
+
+static void startup_on_processor_reset()
```

Comment:
Are one line functions necessary?

Reply:
They are not necessary, but I find them to be more expressive with regards to the requirement fulfillment - think of them as "dedicated startup procedures (named, discrete), filled with dummy implementation (one print)".